src/Lazy/fork.ts

```diff
@@ -0,0 +1,102 @@
+import { isAsyncIterable, isIterable } from "../_internal/utils";
+import isNil from "../isNil";
+import type IterableInfer from "../types/IterableInfer";
+
+type ReturnForkType<A extends Iterable<unknown> | AsyncIterable<unknown>> =
+  A extends AsyncIterable<any>
+    ? AsyncIterableIterator<IterableInfer<A>>
+    : IterableIterator<IterableInfer<A>>;
+
+type Node<T> = { value: T; done?: boolean };
+
+class ForkItem<T> {
+  node: Node<T>;
+  nextNode: ForkItem<T> | null;
+
+  constructor(node: Node<T>) {
+    this.node = node;
+    this.nextNode = null;
+  }
+}
+
+class ForkQueue<T> {
+  head: ForkItem<T>;
+
+  current: ForkItem<T>;
+
+  constructor() {
+    this.head = new ForkItem(null as any);
+    this.current = this.head;
+  }
+
+  toString() {
+    const arr = [];
+    let cur: ForkItem<T> | null = this.head.nextNode;
+    while (cur) {
+      arr.push(cur.node.value);
+      cur = cur.nextNode;
+    }
+
+    return arr.join(", ");
+  }
+}
```

Comment:
It's written as a concrete implementation, which is ForkItem, but it seems to be a queue implemented by LinkedList and LinkedList. Why don't we write it as a separate file and remove the context of Fork so that it can be reused?

Reply:
applied it https://github.com/marpple/FxTS/pull/244/commits/2bbede95fbd8a9f70a63022a1f5d586c50062459 :)

Reply:
LGTM